File: src/main/kotlin/me/varoa/ntrsv/song/Song.kt
package me.varoa.ntrsv.song

import kotlinx.serialization.Serializable
import java.util.*

@Serializable
data class Song(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val artists: List<String>,
    val album: Album,
    val maps: Map<Difficulty, Int>,
    val sortNumber: Int,
    val tags: List<String>,
)

File: src/types/song.ts
import type { Album } from "@/types/album"
import type { Difficulty } from "@/types/difficulty"

export interface Song {
    id: string
    title: string
    artists: string[]
    album: Album
    maps: Partial<Record<Difficulty, number>>
    sortNumber: number
    tags: string[]
}